### PR TITLE
Add TSN support documentation for SBL

### PR DESCRIPTION
**Pull request by sindhuragrandhi**
Add a new page that specifies all the details related to
TSN in SBL.

Signed-off-by: Sindhura Grandhi <sindhura.grandhi@intel.com>

### CODE REVIEW

File: source/how-tos/enable-intel-tsn.rst [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+TSN enable
+--------------------
+
+For each platform, there is a build configuration file BoardConfig.py. User could check this build configuration  file to make sure self.ENABLE_TSN_MAC_ADDRESS is set to 1 to enable TSN MAC Address binary.
+The other TSN related configuration data is PchTsnEnable. Make sure to set this to 1 under Silicon Configuration Data.
+
+.. _tsn_container:
+
+TSN container
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:37]: Can you please move this after overview?

File: source/how-tos/enable-intel-tsn.rst [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+--------------------
+
+Container support is a generic SBL feature. Please refer to |Container| for generic details.
+
+.. |Container| raw:: html
+
+   <a href="https://slimbootloader.github.io/security/container-security.html?highlight=container#container-security" target="_blank">Container</a>
+
+
+SBL defines one such container for TSN binary. The raw TSN MAC binary is packaged as a container in SBL.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:47]: SBL defines a container for IPFW settings. Settings related to TSN is managed as a component inside this IPFW container.

File: source/how-tos/enable-intel-tsn.rst [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+SBL loads this container during Runtime and is processed by Intel® Firmware Support Package (Intel® FSP) during Runtime.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:48]: SBL loads this container during Runtime and processes it as required including passing the relevant settings to Intel® Firmware Support Package

File: source/how-tos/enable-intel-tsn.rst [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    self.ENABLE_TSN_MAC_ADDRESS = 1
+    if self.ENABLE_TSN_MAC_ADDRESS:
+        self.TMAC_SIZE = 0x00001000
+
+    # Name | Image File     | CompressAlg |         AuthType          |                 Key File                                              |Region Align   | Region Size  | Svn Info
+    # ===============================================================================================================================================================
+    ('IPFW',  'SIIPFW.bin',      ''    ,     container_list_auth_type,   'KEY_ID_CONTAINER'+'_'+self._RSA_SIGN_TYPE,                  0,          0                ,         0),   # Container Header
+    ('TMAC',     '             ',     'Lz4',     container_list_auth_type,   'KEY_ID_CONTAINER_COMP'+'_'+self._RSA_SIGN_TYPE,     0,   self.TMAC_SIZE,        0),   # TSN MAC Address
+
+In the build configuration file, user could set ``self.ENABLE_TSN_MAC_ADDRESS`` to 1 to enable TSN container and adjust the size if needed. The TSN container name is``IPFW`` by default.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:62]: to enable TSN component in the IPFW container

File: source/how-tos/enable-intel-tsn.rst [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+Inside each container, the TSN MAC has a unique region name identified by 4 characters. SBL would use it when loading container and update it via firmware update.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:63]: Inside the IPFW container, the TSN MAC component is identified by 4 characters

File: source/how-tos/enable-intel-tsn.rst [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+   <a href="https://www.intel.com/content/www/us/en/industrial-automation/programmable/applications/automation/tsn.html" target="_blank">TSN</a>
+
+
+**TSN MAC Address**
+
+User needs to provide TSN MAC address that is unique per board and that can be independently updateable. FBU tool is used to create a raw TSN MAC binary. Details on how this is packaged and processed will
+be discussed in the later sections.
+
+SBL defines one TSN container inside its BIOS region. This container has the TSN MAC Address binary that can be updated via SBL firmware update mechanism. During boot time, SBL would load TSN MAC binary from the TSN container and process it at Stage 2 during Runtime.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:22]: remove - SBL defines one TSN container inside its BIOS region.

File: source/how-tos/enable-intel-tsn.rst [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+User could generate the raw TSN MAC Address binary with the help of FBU (Firmware and BIOS Utility) Tools. This raw binary need to be signed using the same key used in SBL build so that it could pass SBL verification during boot time.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:24]: is raw binary need to be signed using the component key used during SBL build

File: source/how-tos/enable-intel-tsn.rst [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+For each platform, there is a build configuration file BoardConfig.py. User could check this build configuration  file to make sure self.ENABLE_TSN_MAC_ADDRESS is set to 1 to enable TSN MAC Address binary.
+The other TSN related configuration data is PchTsnEnable. Make sure to set this to 1 under Silicon Configuration Data.
+
+.. _tsn_container:
+
+TSN container
+--------------------
+
+Container support is a generic SBL feature. Please refer to |Container| for generic details.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:40]: Can you please adda a link to the cotainer tool - https://slimbootloader.github.io/tools/GenContainer.html

File: source/how-tos/enable-intel-tsn.rst [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+.. |Container| raw:: html
+
+   <a href="https://slimbootloader.github.io/tools/GenContainer.html" target="_blank">Container</a>
+
+.. |Container Security| raw:: html
+
+   <a href="https://slimbootloader.github.io/security/container-security.html?highlight=container#container-security" target="_blank">Container Security</a>
+
+
+SBL defines a container called IPFW for settings related to TSN and TSN is managed as a component inside this IPFW container.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:33]: SBL defines a container called IPFW for settings related to various silicon IP and settings related to TSN are managed as a component inside this IPFW container.

File: source/how-tos/enable-intel-tsn.rst [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+SBL loads this container and processes it as required including passing the relevant settings to Intel® Firmware Support Package (Intel® FSP) during Runtime.
+SBL defines this container in each platform's build config file BoardConfig.py.
+
+**TSN MAC Address**
+
+User needs to provide TSN MAC address that is unique per board and that can be independently updateable. FBU (Firmware and BIOS Utility) tool is used to create a raw TSN MAC binary.
+This raw TSN binary needs to be signed using the component key that is used during SBL build so that it could pass SBL verification during boot time.
+The signed TSN component is kept into SBL container called IPFW and encapsulate it to create a capsule image. Capsule image is then signed by a firmware update key.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:41]: The signed TSN component is kept into SBL container called IPFW and encapsulate it to create a capsule image. Capsule image is then signed by a firmware update key ==> remove this line??

Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:41]: Remove lines 41 to 48. Same details are presented below in **2. Update using firmware update flow** (line 95)

File: source/how-tos/enable-intel-tsn.rst [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+    self.ENABLE_TSN_MAC_ADDRESS = 1
+    if self.ENABLE_TSN_MAC_ADDRESS:
+        self.TMAC_SIZE = 0x00001000
+
+    # Name | Image File     | CompressAlg |         AuthType          |                 Key File                   | Region Align   | Region Size  | Svn Info
+    # ===============================================================================================================================================================
+    ('IPFW',  'SIIPFW.bin',      ''    ,     container_list_auth_type,   'KEY_ID_CONTAINER'+'_'+self._RSA_SIGN_TYPE,          0,          0      ,        0),   # Container Header
+    ('TMAC',     ''       ,     'Lz4'  ,     container_list_auth_type,   'KEY_ID_CONTAINER_COMP'+'_'+self._RSA_SIGN_TYPE,     0,   self.TMAC_SIZE,        0),   # TSN MAC Address
+
+In the build configuration file, user can set ``self.ENABLE_TSN_MAC_ADDRESS`` to 1 to enable TSN component and adjust the size if needed. The TSN container name is``IPFW`` by default.
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:68]: The TSN container name is``IPFW`` by default.- remove this line. covered already

File: source/how-tos/enable-intel-tsn.rst [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+**1. Specify in Board Config file and rebuild SBL**
+
+
+Specify the raw TSN binary Image file (that is generated by FBU tool) name in build config file BoardConfig.py. Copy the binary into Platform/<Platform>BoardPkg/Binaries folder in SBL source code before building SBL.
+User can also specify different compression and authentication method for
+this binary as desired. Rebuild SBL successfully. Now, the TSN MAC binary will be created and embedded in the IPFW container and be part of the IFWI. When flashed onto the board, SBL will process this container.
+
+In addition, if the user decide to take this option, SBL that was rebuilt need to be stitched with TSN stitching option.
+
+**TSN Stitch Option for Soft Straps**
```
Comment (rprangar) [inline source/how-tos/enable-intel-tsn.rst:85]: Move lines 85 to 93 to 101 (after .. _tsn_settings: and before TSN configuration settings)?

Comment (sindhuragrandhi) [reply source/how-tos/enable-intel-tsn.rst:85]: The flow is much better if this section is tagged along in option 1 method as it doesnt pertain to option 2. Leaving it here for now.

**Review by rprangar — APPROVED** [review]
Approved

**Pull request merged**